### PR TITLE
Move PCS to west us2, add disk

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3180
This PR updates the bicep script to create the service in WestUS2. It also makes the service used Managed Identity instead of App Principle.
And it adds the 50gb disk that we will use for the VMR.

The bicep template has already been used to create the new staging environment.

P.S managed identity client id is not a secret

### Release Note Category
- [x] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description
Move PCS to WestUS2, use Managed Identity and add disk

### CODE REVIEW

File: eng/service-templates/ProductConstructionService/provision.bicep [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
 @description('Name of the Azure Container Registry resource into which container images will be published')
 param containerRegistryName string = 'productconstructionint'
 
 @description('CPU cores allocated to a single container instance, e.g., 0.5')
-param containerCpuCoreCount string = '0.25'
+param containerCpuCoreCount string = '0.5'
 
 @description('Memory allocated to a single container instance, e.g., 1Gi')
-param containerMemory string = '0.5Gi'
+param containerMemory string = '1Gi'
```
Comment (premun) [inline eng/service-templates/ProductConstructionService/provision.bicep:14]: I think we should do at least 2Gi. Git can take quite some memory when working with large repositories.
LibGit2Sharp takes even more.

```suggestion
param containerMemory string = '2Gi'
```

File: eng/service-templates/ProductConstructionService/provision.bicep [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
+param location string = 'westus2'
 
 @minLength(5)
 @maxLength(50)
 @description('Name of the Azure Container Registry resource into which container images will be published')
 param containerRegistryName string = 'productconstructionint'
 
 @description('CPU cores allocated to a single container instance, e.g., 0.5')
-param containerCpuCoreCount string = '0.25'
+param containerCpuCoreCount string = '0.5'
```
Comment (premun) [inline eng/service-templates/ProductConstructionService/provision.bicep:11]: Let's do `1.0`

```suggestion
param containerCpuCoreCount string = '1.0'
```

File: eng/service-templates/ProductConstructionService/provision.bicep [hunk 3]
```diff
@@ -96,28 +104,31 @@ resource containerRegistry 'Microsoft.ContainerRegistry/registries@2022-02-01-pr
...
-// }
+
+resource identity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: identityName
+  location: location
+}
+
+var principalId = identity.properties.principalId
+
+// azure system role for setting up acr pull access
```
Comment (premun) [inline eng/service-templates/ProductConstructionService/provision.bicep:115]: Great that you added the comments

File: eng/service-templates/ProductConstructionService/provision.bicep [hunk 4]
```diff
@@ -161,13 +172,17 @@ var env = [
...
-  properties: {
-      managedEnvironmentId: containerAppsEnvironment.id
-      configuration: {
+// container app hosting the Product Construction Service
+resource containerapp 'Microsoft.App/containerApps@2023-04-01-preview' = {
+    name: productConstructionServiceName
+    location: location
+    identity: {
+            type: 'UserAssigned'
+            userAssignedIdentities: { '${identity.id}' : {}}
```
Comment (premun) [inline eng/service-templates/ProductConstructionService/provision.bicep:181]:
```suggestion
        type: 'UserAssigned'
        userAssignedIdentities: { '${identity.id}' : {}}
```

File: eng/service-templates/ProductConstructionService/provision.bicep [hunk 5]
```diff
@@ -176,57 +191,59 @@ resource apiservice 'Microsoft.App/containerApps@2023-04-01-preview' = {
...
-                            path: '/health'
-                            port: 8080
-                            scheme: 'HTTP'
-                        }
-                        initialDelaySeconds: 5
-                        periodSeconds: 10
-                        successThreshold: 1
-                        failureThreshold: 3
-                        type: 'Startup'
+        template: {
```
Comment (premun) [inline eng/service-templates/ProductConstructionService/provision.bicep:201]: Somehow this got a wrong indent by 2 spaces?

File: eng/service-templates/ProductConstructionService/provision.bicep (hunk 1 above)
Comment (dkurepa) [reply eng/service-templates/ProductConstructionService/provision.bicep:14]: done, we can increase it even further. We'll also have to increase the CPU core count tho https://learn.microsoft.com/en-us/azure/container-apps/containers#:~:text=Consumption%20workload%20profile-,0.25,2.5Gi,-%E2%9C%94

File: eng/service-templates/ProductConstructionService/provision.bicep (hunk 5 above)
Comment (dkurepa) [reply eng/service-templates/ProductConstructionService/provision.bicep:201]: it was wrong before, I fixed it in this PR

Comment (premun) [reply eng/service-templates/ProductConstructionService/provision.bicep:201]: image

Should it not be aligned?

Like here?

image


Just asking - don't know bicep

File: eng/service-templates/ProductConstructionService/provision.bicep (hunk 1 above)
Comment (premun) [reply eng/service-templates/ProductConstructionService/provision.bicep:14]: Yeah, I think 2 and 8 might be feasible eventually but let's revisit later. For now, I am pretty sure we'd hit 1Gi, 2Gi should be okay

**Review by premun — APPROVED** [review]

File: eng/service-templates/ProductConstructionService/provision.bicep (hunk 5 above)
Comment (dkurepa) [reply eng/service-templates/ProductConstructionService/provision.bicep:201]: Hm, it doesn't look like that in my vscode, that's weird

Comment (dkurepa) [reply eng/service-templates/ProductConstructionService/provision.bicep:201]: think I fixed it now

**Comment by oleksandr-didyk** [comment]
Rolled out by https://github.com/dotnet/arcade-services/issues/3264

**Pull request merged**